src/main/java/uk/gov/companieshouse/api/accounts/service/impl/ProfitAndLossService.java

```diff
@@ -195,15 +202,27 @@ private void initLinks(ProfitAndLoss profitAndLoss, String link) {
         profitAndLoss.setLinks(map);
     }
 
-    private String generateID(String companyAccountId, HttpServletRequest request) {
+    private String generateID(String companyAccountId, AccountingPeriod period) {
 
         return keyIdGenerator.generate(companyAccountId + "-" +
-                (isCurrentPeriod(request) ? ResourceName.CURRENT_PERIOD.getName() : ResourceName.PREVIOUS_PERIOD.getName()) +
+                (isCurrentPeriod(period) ? ResourceName.CURRENT_PERIOD.getName() : ResourceName.PREVIOUS_PERIOD.getName()) +
                 "-" + ResourceName.PROFIT_LOSS.getName());
     }
 
-    private boolean isCurrentPeriod(HttpServletRequest request) {
+    private boolean isCurrentPeriod(AccountingPeriod period) {
+
+        return period.equals(AccountingPeriod.CURRENT_PERIOD);
+    }
+
+    private void invalidateStatementsIfExisting(String companyAccountId, HttpServletRequest request)
+        throws DataException {
+
+        Statement statement = new Statement();
```

Comment:
Should this logic exist within the if? We may create a `Statement` object and never use it if the find returns a NOT_FOUND status. Same with the retrieval of the transaction from the request on the line below. We also need to set `hasAgreedToLegalStatements` to false on the `statement` object.